web-components/src/components/inputs/new/ConfirmationCode/ConfirmationCode.tsx

```diff
@@ -0,0 +1,189 @@
+    }, [autoFocus]);
+
+    const sendResult = () => {
+      const res = inputsRef.current.map(input => input.value).join('');
+      onChange && onChange(res);
+    };
+
+    const handleOnChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+      const {
+        target: { value, nextElementSibling },
+      } = e;
+      if (value.length > 1) {
+        e.target.value = value.charAt(0);
+        if (nextElementSibling !== null) {
+          (nextElementSibling as HTMLInputElement).focus();
+        }
+      } else {
+        if (value.match(inputProps.pattern)) {
+          if (nextElementSibling !== null) {
+            (nextElementSibling as HTMLInputElement).focus();
+          }
+        } else {
+          e.target.value = '';
+        }
+      }
+      sendResult();
+    };
+
+    const handleOnKeyDown = (e: React.KeyboardEvent<HTMLInputElement>) => {
+      const { key } = e;
+      const target = e.target as HTMLInputElement;
+      if (key === 'Backspace') {
+        if (target.value === '') {
+          if (target.previousElementSibling !== null) {
+            const t = target.previousElementSibling as HTMLInputElement;
+            t.value = '';
+            t.focus();
+            e.preventDefault();
+          }
+        } else {
+          target.value = '';
+        }
+        sendResult();
+      }
+    };
+
+    const handleOnFocus = (e: React.FocusEvent<HTMLInputElement>) => {
+      e.target.select();
+    };
+
+    const handleOnPaste = (e: React.ClipboardEvent<HTMLInputElement>) => {
+      const pastedValue = e.clipboardData.getData('Text');
+
+      let currentInput = 0;
+
+      for (let i = 0; i < pastedValue.length; i++) {
+        const pastedCharacter = pastedValue.charAt(i);
+        const currentValue = inputsRef.current[currentInput].value;
+        if (pastedCharacter.match(inputProps.pattern)) {
+          if (!currentValue) {
+            inputsRef.current[currentInput].value = pastedCharacter;
+            if (inputsRef.current[currentInput].nextElementSibling !== null) {
+              (
+                inputsRef.current[currentInput]
+                  .nextElementSibling as HTMLInputElement
+              ).focus();
+              currentInput++;
+            }
+          }
+        }
+      }
+      sendResult();
+
+      e.preventDefault();
+    };
+
+    const inputs = [];
+    for (let i = 0; i < length; i++) {
+      inputs.push(
+        <input
+          key={i}
+          onChange={handleOnChange}
+          onKeyDown={handleOnKeyDown}
+          onFocus={handleOnFocus}
+          onPaste={handleOnPaste}
+          {...inputProps}
+          type={isPassword ? 'password' : inputProps.type}
+          ref={(el: HTMLInputElement) => {
+            inputsRef.current[i] = el;
+          }}
+          maxLength={1}
+          className={cn(
+            'w-[45px] h-[60px] bg-grey-0 border-grey-300 border mr-5 sm:mr-20 text-lg text-center p-0 font-lato focus:outline-0 rounded-sm border-solid',
```

Comment:
Actually, 60 was already there. Regarding 45 I think we should not add all custom values otherwise the spacing scale loses a bit of its purpose. That being said we already have values that are a multiple of 5 in the scale so maybe let's use that as references to decide what should be in scale or not (multiplier of 5 above 10), wdyt?